### PR TITLE
Start of game implementation

### PR DESCRIPTION
**Pull request by AndreasLrx**
Add a cube map, with a cubic controlable player (in fact it isn't cubic, **it is** a cube) and collision resolutions between cubes... Starts to looks like minecraft here.

Most of added components are not yet used, this PR is the first mandatory part of the game before it starts to looks like a real game.

Linked: #110
Requires: #113

### CODE REVIEW

**Review by ZQUEMA — APPROVED** [review]

File: src/game/Game.cpp [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+                .with<components::Position>(cell.x, 1.f, cell.y)
+                .with<components::Velocity>()
+                .with<components::Living>(_params.livesCount)
+                .with<components::Collidable>()
+                .with<components::Player>()
+                .with<components::Cube>()
+                .with<components::Size>(0.7f, 2.f, 0.7f)
+                .with<components::CubeColor>(raylib::core::Color::RED)
+                .with<components::Controlable>(owner,
+                    [this](ecs::Entity self, ecs::SystemData data, const Users::ActionEvent &event) {
```
Comment (MisterPeModder) [inline src/game/Game.cpp:74]: Moving this huge lambda into a separate function would make this a lot more readable.
Something like: `.with<...>(owner, &World::processEntityAction)`

File: src/game/components/Destructible.hpp [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+#define GAME_COMPONENTS_DESTRUCTIBLE_HPP_
+
+#include "ecs/Component.hpp"
+
+namespace game::components
+{
+    /// Destructible component
+    struct Destructible final : public ecs::Component {
+        /// If the component is destroyed.
+        bool destructed;
```
Comment (MisterPeModder) [inline src/game/components/Destructible.hpp:18]: `destroyed` is the word.

File: src/game/systems/Collision.cpp [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+namespace game::systems
+{
+    void Collision::run(ecs::SystemData data)
+    {
+        auto &positions = data.getStorage<Position>();
+        auto &sizes = data.getStorage<Size>();
+        auto &entities = data.getResource<ecs::Entities>();
+        auto &collidable = data.getStorage<Collidable>();
+        auto &velocities = data.getStorage<Velocity>();
+        auto optionnalVelocity = ecs::maybe(velocities);
```
Comment (MisterPeModder) [inline src/game/systems/Collision.cpp:25]: optional is written with one `n`.

**Review by MisterPeModder — CHANGES_REQUESTED** [review]
Just little things to change, otherwise looks (and works) pretty good.
The Web is broken though, I suspect PR #109 might fix this, so I will ignore that.

File: src/game/Game.cpp (hunk 1 above)
Comment (AndreasLrx) [reply src/game/Game.cpp:74]: I have planned to do this for the next part of the game implementation.

**Review by MisterPeModder — APPROVED** [review]
Fix your conficts with the main() and we're good

**Pull request merged**